Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -1443,6 +1459,38 @@ void HTMLInputElement::apply_presentational_hints(CSS::StyleProperties& style) c
     });
 }
 
+// https://html.spec.whatwg.org/multipage/input.html#attr-input-maxlength
+WebIDL::Long HTMLInputElement::max_length() const
+{
+    // The maxLength IDL attribute must reflect the maxlength content attribute, limited to only non-negative numbers.
+    if (auto maxlength_string = get_attribute(HTML::AttributeNames::maxlength); maxlength_string.has_value()) {
+        if (auto maxlength = parse_non_negative_integer(*maxlength_string); maxlength.has_value())
+            return *maxlength;
+    }
+    return -1;
+}
+
+WebIDL::ExceptionOr<void> HTMLInputElement::set_max_length(WebIDL::Long value)
+{
+    return set_attribute(HTML::AttributeNames::maxlength, MUST(String::number(value)));
```

Comment:
These setters are all supposed to throw an exception if a bad value is provided:
https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#limited-to-only-non-negative-numbers
> If the reflected IDL attribute is limited to only non-negative numbers and the given value is negative, then throw an "IndexSizeError" DOMException.

Reply:
Instead of repeating all this parsing code (which will get longer with that step implemented), let's move them to AOs with that spec link implemented. Then we can just call them from the input/text elements.

Reply:
What do you mean with "move them to AOs"?

Reply:
Abstract Operations are basically the name the JS specs use for shared algorithms. So, just moving the parsing logic into a separate, shared function somewhere. IIRC we already have one or two somewhere, for parsing numeric attributes.

Reply:
I've created `convert_non_negative_integer_to_string()`